doc/connectivity-semantics-and-api.md

```diff
@@ -75,6 +90,15 @@ corresponding reasons. Empty cells denote disallowed transitions.
     <td></td>
     <td></td>
     <td></td>
+    <td>Shutdown triggered by application.</td>
+  </tr>
+  <tr>
+    <th>IDLE</th>
+    <td>Any new RPC activity on the channel</td>
```

Comment:
A READY channel that GETS a GOAWAY but has pending RPCs should go to TRANSIENT_FAILURE and then CONNECTING.

Channels with pending RPCs should never enter IDLE as by definition, they are not idle and have some work to do.